crates/shared/src/current_block/get_block_and_call.rs

```diff
@@ -1,8 +1,9 @@
 use {
```

Comment:
I would rename the module as well `get_block_and_call` is not helpful IMO here (since we no longer have different strategies).

Maybe just `retriever.rs` or `fetcher.rs` to indicate its the logic for the `current_block::{retriever,fetcher}`?